src/darwin/Framework/CHIP/MTRCSRInfo.h

```diff
@@ -20,9 +20,31 @@
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- * Represents information relating to NOC CSR.
+ * Represents information relating to a certificate signing request for a Matter
+ * operational certificate.  The entire TLV nocsr-elements structure, including
+ * any vendor-specific data, is included, as well as the parsed-out CSR and
+ * CSRNonce fields from that structure.
  *
+ * The AttestationSignature from the CSRResponse command is also included.  This
+ * signs a concatenation of csrElements and the attestationChallenge found in
+ * MTRAttestationInfo.
  */
+MTR_NEWLY_AVAILABLE
+@interface MTROperationalCSRInfo : NSObject
+
+@property (nonatomic, copy, readonly) NSData * csr;
+@property (nonatomic, copy, readonly) NSData * csrNonce;
+@property (nonatomic, copy, readonly) NSData * csrElements;
```

Comment:
It would be nice if it was more obvious which is the overall TLV structure... maybe this property could be called `rawTLV` or something along those lines? Also would be nice if we had a generic `MTRTLVBytes` type.

Reply:
MTRTLVBytes makes a lot of sense.  Will add that, and add better docs here, along with possible renaming as needed.

Reply:
Done.